src/View/Helper/FormHelper.php

```diff
@@ -1939,7 +1963,20 @@ public function postLink(string $title, array|string|null $url = null, array $op
         } else {
             $onClick .= ' event.returnValue = false; return false;';
         }
-        $options['onclick'] = $onClick;
+
+        if ($this->_View->getRequest()->getAttribute('cspScriptNonce')) {
+            $options['id'] ??= $this->_domId('link-' . $formName);
+            $script = $this->templater()->format('postLinkJs', [
+                'linkId' => $options['id'],
+                'content' => $onClick,
+            ]);
+
+            $out .= $this->Html->scriptBlock($script, ['block' => $options['block']]);
```

Comment:
This is pretty clever :clap: I like that the CSP middleware and formhelper are working better together. I could see this being a pattern we could use for generated confirm messages as well.